src/test/dispatcher.ts

```diff
@@ -299,7 +339,7 @@ export class Dispatcher {
     if (!maxFailures || this._failureCount <= maxFailures)
       this._reporter.onTestEnd(test, result);
     if (maxFailures && this._failureCount === maxFailures)
-      this._isStopped = true;
+      this.stop().catch(e => {});
```

Comment:
Can we make sure `stop` is idempotent. It looks like we can send multiple `stop` messages to a worker with this patch.